proto/def/contract.proto

```diff
@@ -0,0 +1,87 @@
+syntax = 'proto3';
+
+import "google/protobuf/empty.proto";
+
+option go_package = "control-plane/pkg/contract";
+option java_package = "dev.knative.eventing.kafka.broker.contract";
+option java_outer_classname = "DataPlaneContract";
+
+message Filter {
+  // attributes filters events by exact match on event context attributes.
+  // Each key in the map is compared with the equivalent key in the event
+  // context. An event passes the filter if all values are equal to the
+  // specified values.
+  //
+  // Nested context attributes are not supported as keys. Only string values are supported.
+  map<string, string> attributes = 1;
+}
+
+message Egress {
+  // consumer group name
+  string consumerGroup = 1;
+
+  // destination is the sink where events are sent.
+  string destination = 2;
+
+  oneof replyStrategy {
+    // Send the response to an url
+    string replyUrl = 3;
+
+    // Send the response to a Kafka topic
+    google.protobuf.Empty replyToOriginalTopic = 4;
+  }
+
+  // Dead letter is where the event is sent when something goes wrong
+  string deadLetter = 5;
```

Comment:
The deadLetterSink should be in `Ingress` or in `Resource` otherwise as [you can see](https://github.com/knative-sandbox/eventing-kafka-broker/pull/162/files#diff-c424bcd353310ce810d301f87b32ce92R903) we need to duplicate this information for all Triggers and the broker reconciler needs to update it for all triggers, so it cost O(N + M) instead of O(N) only + the wasted space.

Reply:
The reason for the change is:

* It doesn't make sense to configure deadLetter for the ingress, because it's a specification of what to do on the "dispatch" side, not on receiver side. It may make sense to have it in Resource, but it's more a "resource wise egress config"
* We need it on the egress side in order to support in future the channel impl, because in channels every sub has its own deadletter

Reply:
As I said I'm worried about wasting space, so for me, a DLS in the resource makes sense as resource-wise egrees config